verilog/tools/ls/verilog-language-server.cc

```diff
@@ -88,66 +120,45 @@ void VerilogLanguageServer::SetRequestHandlers() {
         return verilog::FormatRange(
             parsed_buffers_.FindBufferTrackerOrNull(p.textDocument.uri), p);
       });
+  dispatcher_.AddRequestHandler(  // go-to definition
+      "textDocument/definition",
+      [this](const verible::lsp::DefinitionParams &p) {
+        return symbol_table_handler_.FindDefinition(p, parsed_buffers_);
+      });
   // The client sends a request to shut down. Use that to exit our loop.
   dispatcher_.AddRequestHandler("shutdown", [this](const nlohmann::json &) {
     shutdown_requested_ = true;
     return nullptr;
   });
 }
 
-absl::Status VerilogLanguageServer::Run(const ReadFun &read_fun) {
-  shutdown_requested_ = false;
-  absl::Status status = absl::OkStatus();
-  while (status.ok() && !shutdown_requested_) {
-    status = stream_splitter_.PullFrom(read_fun);
+verible::lsp::InitializeResult VerilogLanguageServer::InitializeRequestHandler(
+    const verible::lsp::InitializeParams &p) {
+  // set VerilogProject for the symbol table, if possible
+  if (!p.rootUri.empty()) {
+    absl::string_view path = verilog::LSPUriToPath(p.rootUri);
+    if (path.empty()) {
+      LOG(ERROR) << "Unsupported rootUri in initialize request:  " << p.rootUri
+                 << std::endl;
+      path = p.rootUri;
+    }
+    ConfigureProject(path);
+  } else if (!p.rootPath.empty()) {
+    ConfigureProject(p.rootPath);
```

Comment:
If neither `p.rootUri` is set or `p.rootPath`, we never call `ConfigureProject()` and end up with a `nullptr` project.

Reply:
Probably a good fallback would be to call ConfigureProject() with the directory where `verible.filelist` was found. And if _that_ doesn't exist, a fallback of `ConfigureProject(".")` probably.

But just the `LOG(WARNING)` below leaves the project unconfigured, so no symbol table functionality will be available.